firewood/src/storage/linear/mod.rs

```diff
@@ -96,3 +96,99 @@ impl<S: ReadLinearStore> ReadLinearStore for LinearStore<S> {
         todo!()
     }
 }
+
+pub mod tests {
+    use super::{ReadLinearStore, WriteLinearStore};
+    use std::io::Read;
+
+    #[derive(Debug)]
+    pub struct InMemReadWriteLinearStore {
+        bytes: Vec<u8>,
+    }
+
+    impl InMemReadWriteLinearStore {
+        pub fn new() -> Self {
+            Self { bytes: vec![] }
+        }
+    }
+
+    impl WriteLinearStore for InMemReadWriteLinearStore {
+        fn write(&mut self, offset: u64, object: &[u8]) -> Result<usize, std::io::Error> {
+            let offset = offset as usize;
+            if offset + object.len() > self.bytes.len() {
+                self.bytes.resize(offset + object.len(), 0);
+            }
+            self.bytes[offset..offset + object.len()].copy_from_slice(object);
+            Ok(object.len())
+        }
+    }
+
+    impl ReadLinearStore for InMemReadWriteLinearStore {
+        fn stream_from(&self, addr: u64) -> Result<impl Read, std::io::Error> {
+            Ok(&self.bytes[addr as usize..])
```

Comment:
This can panic. I know it's for testing, but it should probably be consistent with how the real one works (assuming the real one won't panic)